KuduSync.NET/Program.cs

```diff
@@ -7,20 +7,53 @@ namespace KuduSync.NET
 {
     class Program
     {
+        public const string AppOfflineFileName = "app_offline.htm";
+        private const string AppOfflineFileContent =
+            "<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\" \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">" +
+            "<html xmlns=\"http://www.w3.org/1999/xhtml\" > " +
+            "<head> " +
+            "    <title>Site Under Construction</title> " +
+            "</head> " +
+            "<body>" +
+            "<!--                                                                                                                        " +
+            "                                                                                                                            " +
+            "    Adding additional hidden content so that IE Friendly Errors don't prevent                                               " +
+            "    this message from displaying (note: it will show a 'friendly' 404                                                       " +
+            "    error if the content isn't of a certain size).                                                                          " +
+            "                                                                                                                            " +
+            "-->" +
+            "</body>" +
+            "</html>";
+        private const string AppOfflineSetting = "KUDU_APP_OFFLINE_CREATION";
+
         static int Main(string[] args)
         {
             var stopwatch = Stopwatch.StartNew();
             var kuduSyncOptions = new KuduSyncOptions();
             int exitCode = 0;
-
+            var appOfflineCreated = false;
+            var appOfflineSetting = Environment.GetEnvironmentVariable(AppOfflineSetting);
             try
             {
                 ICommandLineParser parser = new CommandLineParser();
                 if (parser.ParseArguments(args, kuduSyncOptions))
                 {
                     using (var logger = GetLogger(kuduSyncOptions))
                     {
-                        new KuduSync(kuduSyncOptions, logger).Run();
+                        // The default behavior is to create the app_offline.htm page
+                        if (string.IsNullOrWhiteSpace(appOfflineSetting) || !appOfflineSetting.Equals("false", StringComparison.OrdinalIgnoreCase))
```

Comment:
We typically support the value `1` to turn on (see [other settings](https://github.com/projectkudu/kudu/wiki/Configurable-settings#repository-and-deployment-related-settings))

Reply:
Sounds good, so now users will have to set `SCM_CREATE_APP_OFFLINE` to `0`, if they don't want the default behavior. Let me know if you prefer something like `SCM_DO_NOT_CREATE_APP_OFFLINE` to `1`.